samples/client/petstore/C/api/UserAPI.c

```diff
@@ -0,0 +1,319 @@
+    localVarBodyParameters = cJSON_Print(UserJSONObject);
+
+   apiClient_invoke(apiClient,
+                    "pet",
+                    localVarPath,
+                    localVarQueryParameters,
+                    localVarHeaderParameters,
+                    localVarFormParameters,
+                    localVarBodyParameters,
+                    "POST");
+
+   free(apiClient->dataReceived);
+    free(localVarBodyParameters);
+   cJSON_Delete(UserJSONObject);
+   return;
+
+}
+
+// Creates list of users with given input array
+//
+void *UserAPI_createUsersWithListInput(apiClient_t *apiClient, list_t* User) {
+    list_t    *localVarQueryParameters,
+    list_t    *localVarHeaderParameters,
+    list_t    *localVarFormParameters,
+    char      *localVarBodyParameters,
+
+    // create the path
+   char *localVarPath = malloc(MAX_BUFFER_LENGTH);
+   snprintf(localVarPath, MAX_BUFFER_LENGTH, "/user/createWithList");
+
+
+
+    // JSON HTTP Request - user
+    cJSON *UserJSONObject;
+    UserJSONObject = list_convertToJSON(User);
+    localVarBodyParameters = cJSON_Print(UserJSONObject);
+
+   apiClient_invoke(apiClient,
+                    "pet",
+                    localVarPath,
+                    localVarQueryParameters,
+                    localVarHeaderParameters,
+                    localVarFormParameters,
+                    localVarBodyParameters,
+                    "POST");
+
+   free(apiClient->dataReceived);
+    free(localVarBodyParameters);
+   cJSON_Delete(UserJSONObject);
+   return;
+
+}
+
+// Delete user
+//
+// This can only be done by the logged in user.
+//
+void *UserAPI_deleteUser(apiClient_t *apiClient, char* Username) {
+    list_t    *localVarQueryParameters,
+    list_t    *localVarHeaderParameters,
+    list_t    *localVarFormParameters,
+    char      *localVarBodyParameters,
+
+    // create the path
+   char *localVarPath = malloc(MAX_BUFFER_LENGTH);
+   snprintf(localVarPath, MAX_BUFFER_LENGTH, "/user/{username}");
+
+    // TODO path parameter Username (username) not yet supported
+    // TODO base path = http://petstore.swagger.io/v2
+    char* baseNameMod = malloc(strlen(username)+2); //baseNameMod free not yet implemented
+    snprintf(baseNameMod, strlen(baseName)+3, "%s%s%s", "{", username, "}");
+    localVarPath = strReplace(localVarPath, baseNameMod, Username);
+
+
+   apiClient_invoke(apiClient,
+                    "pet",
+                    localVarPath,
+                    localVarQueryParameters,
+                    localVarHeaderParameters,
+                    localVarFormParameters,
+                    localVarBodyParameters,
+                    "DELETE");
+
+   free(apiClient->dataReceived);
+   free(UsernameString);
+   return;
+
+}
+
+// Get user by user name
+//
+user_t *UserAPI_getUserByName(apiClient_t *apiClient, char* Username) {
+    list_t    *localVarQueryParameters,
+    list_t    *localVarHeaderParameters,
+    list_t    *localVarFormParameters,
```

Comment:
How are these used?

Reply:
These variables are later passed to`apiClient_invoke`:  https://github.com/OpenAPITools/openapi-generator/pull/516/files#diff-5632e1467c11db52bdf6ca9036c23f3dR34

Reply:
What is the purpose of them?